src/Chapter4.hs

```diff
@@ -611,6 +644,11 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+instance Monad List where
+    (>>=) :: List a -> (a -> List b) -> List b
+    Empty >>= _     = Empty
+    (Cons x xs) >>= f = extend' (f x) (xs >>= f)
```

Comment:
Here you actually can reuse `concat'` and `Functor` instance for the list to implement this function without explicit pattern matching inside `>>=`